src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```diff
@@ -160,7 +160,11 @@ public function getKernel()
      */
     public function getSurrogate()
     {
-        return $this->getEsi();
+        if (!$this->surrogate instanceof Esi) {
+            throw new \LogicException('This instance of HttpCache was not set up to use ESI as surrogate handler. You must overwrite and use createSurrogate');
+        }
```

Comment:
hum, maybe not: previoulsy, only ESI was allowed (by calling getEsi())
So allowing more surrogate would be a new feature.